File: frontend/src/types/sudo.ts
export interface SudoUserCreate {
  account_id: number;
  tg_user_id: number;
  display_name?: string;
  allowed_chat_ids?: number[];
  allowed_commands?: string[];
}

export interface SudoUserUpdate {
  display_name?: string;
  allowed_chat_ids?: number[];
  allowed_commands?: string[];
}

export interface SudoUserResponse {
  id: number;
  account_id: number;
  tg_user_id: number;
  display_name?: string;
  allowed_chat_ids?: number[];
  allowed_commands?: string[];
  created_at: string;
}

File: backend/app/schemas/sudo.py
"""Sudo 用户管理 Schemas。"""

from __future__ import annotations

from pydantic import BaseModel


class SudoUserCreate(BaseModel):
    """创建 Sudo 用户请求。"""
    account_id: int
    tg_user_id: int
    display_name: str | None = None
    allowed_chat_ids: list[int] | None = None
    allowed_commands: list[str] | None = None


class SudoUserUpdate(BaseModel):
    """更新 Sudo 用户请求。"""
    display_name: str | None = None
    allowed_chat_ids: list[int] | None = None
    allowed_commands: list[str] | None = None


class SudoUserResponse(BaseModel):
    """Sudo 用户响应。"""
    id: int
    account_id: int
    tg_user_id: int
    display_name: str | None = None
    allowed_chat_ids: list[int] | None = None
    allowed_commands: list[str] | None = None
    created_at: str

    class Config:
        from_attributes = True
